core/src/test/java/org/apache/calcite/test/MaterializedViewRelOptRulesTest.java

```diff
@@ -338,6 +341,104 @@ protected final MaterializedViewFixture sql(String materialize,
+            + "from \"events\""
+            + " where \"ts\" >= TIMESTAMP'2018-01-01 00:02:00' "
+            + "AND \"ts\" < TIMESTAMP'2018-01-01 00:05:00'"
+            + "group by \"eventid\", floor(\"ts\" to minute)")
+        .checkingThatResultContains("EnumerableCalc(expr#0..2=[{inputs}], "
+            + "expr#3=[Sarg[[2018-01-01 00:02:00..2018-01-01 00:05:00)]], expr#4=[SEARCH($t1, $t3)"
+            + "], proj#0..2=[{exprs}], $condition=[$t4])\n"
+            + "  EnumerableTableScan(table=[[hr, MV0]])")
+        .withMetadataProvider(
+            ChainedRelMetadataProvider.of(
+                ImmutableList.of(TestMetadataHandlers.TestRelMdRowCount.SOURCE,
+                    TestMetadataHandlers.TestRelMdSelectivity.SOURCE,
+                    DefaultRelMetadataProvider.INSTANCE))
+        )
+        .ok();
+  }
+
+  @Test void testAggregateMaterializationAggregateFuncsRange2() {
+    // if range predicate doesn't align on the rollup column boundary verify that a union on both
+    // the table and view is used
+    sql("select \"eventid\", floor(\"ts\" to minute), count(*) "
+            + "from \"events\""
+            + " group by \"eventid\", floor(\"ts\" to minute)",
+        "select \"eventid\", floor(\"ts\" to minute), count(*) \n"
+            + "from \"events\""
+            + " where \"ts\" > TIMESTAMP'2018-01-01 00:02:30.123' "
+            + "AND \"ts\" <= TIMESTAMP'2018-01-01 00:05:30.456'"
+            + "group by \"eventid\", floor(\"ts\" to minute)")
+        .checkingThatResultContains("EnumerableAggregate(group=[{0, 1}], EXPR$2=[$SUM0($2)])\n"
+            + "  EnumerableUnion(all=[true])\n"
+            + "    EnumerableAggregate(group=[{0, 1}], EXPR$2=[COUNT()])\n"
+            + "      EnumerableCalc(expr#0..1=[{inputs}], expr#2=[FLAG(MINUTE)], expr#3=[FLOOR($t1,"
+            + " $t2)], expr#4=[Sarg[(2018-01-01 00:02:30.123:TIMESTAMP(3)..2018-01-01 "
+            + "00:03:00:TIMESTAMP(3)), [2018-01-01 00:05:00:TIMESTAMP(3)..2018-01-01 00:05:30"
+            + ".456:TIMESTAMP(3)]]:TIMESTAMP(3)], expr#5=[SEARCH($t1, $t4)], eventid=[$t0], "
+            + "$f1=[$t3], $condition=[$t5])\n"
+            + "        EnumerableTableScan(table=[[hr, events]])\n"
+            + "    EnumerableCalc(expr#0..2=[{inputs}], expr#3=[Sarg[[2018-01-01 00:03:00."
+            + ".2018-01-01 00:05:00)]], expr#4=[SEARCH($t1, $t3)], proj#0..2=[{exprs}], "
+            + "$condition=[$t4])\n"
+            + "      EnumerableTableScan(table=[[hr, MV0]])")
+        .withMetadataProvider(
+            ChainedRelMetadataProvider.of(
+                ImmutableList.of(TestMetadataHandlers.TestRelMdRowCount.SOURCE,
+                    TestMetadataHandlers.TestRelMdSelectivity.SOURCE,
+                    DefaultRelMetadataProvider.INSTANCE))
+        )
+        .ok();
+  }
+
+  @Test void testAggregateMaterializationAggregateFuncsRange31() {
+    // test using multiple views
```

Comment:
can you improve the comment to match the explanation of the tests above which is of great help? Notably, please cover why `mv0` is picked over `mv1` and sketch what happens when multiple views are available

Reply:
Added a more descriptive comment.